Sources/ClientRuntime/Networking/Http/HttpClientConfiguration.swift

```diff
@@ -9,11 +9,18 @@ import struct Foundation.TimeInterval
 
 public class HttpClientConfiguration {
 
-    /// The timeout for a request, in seconds.
+    /// The timeout for establishing a connection, in seconds.
     ///
     /// If none is provided, the client will use default values based on the platform.
     public var connectTimeout: TimeInterval?
 
+    /// The timeout for socket, in seconds.
+    /// Sets maximum time to wait between two data packets.
+    /// Used to close stale connections that have no activity.
+    ///
+    /// If no value is provided, the defaut client won't have a socket timeout.
+    public var socketTimeout: TimeInterval?
```

Comment:
URLSession does not have separate settings for timeout on initial connection, and for idle time between data transmissions on an existing connection.

https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1408259-timeoutintervalforrequest

seems to be the setting for both and the two cannot be set individually.
Our doc comments should make clear which settings apply to which HTTP client and which are ignored.

Reply:
Found a separate setting for initial connection timeout in URLRequest: https://developer.apple.com/documentation/foundation/urlrequest/2011509-timeoutinterval/

connectTimeout config on HttpClientConiguration is now set for URLRequest:timeoutInterval.